File: Projects/Graffer/0.2/grafer/ui/MainWindow.py
'''	
	This file is part of Camarillo.
	
	Copyright (C) 2008	Frederic-Gerald Morcos <fred.morcos@gmail.com>
	
	Camarillo is free software: you can redistribute it and/or modify
	it under the terms of the GNU General Public License as published by
	the Free Software Foundation, either version 3 of the License, or
	(at your option) any later version.

	Camarillo is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with Camarillo.	If not, see <http://www.gnu.org/licenses/>.
'''

from gtk import Window
from MainVBox import MainVBox
import Callbacks

class MainWindow (Window):
	def __init__(self):
		Window.__init__(self)
		
		self.aboutDialog = None
		self.mainVBox = MainVBox()
		
		self.set_title('Grafer')
		self.set_border_width(2)
		self.add(self.mainVBox)
		
		self.connect('destroy', Callbacks.mainWindow_destroy, None)
		self.mainVBox.mainMenu.menuAbout.connect(
				'activate', Callbacks.menuAbout_activate, self)
		self.mainVBox.mainMenu.menuNew.connect(
				'activate', Callbacks.menuNew_activate, self)
		self.mainVBox.mainMenu.menuClose.connect(
				'activate', Callbacks.menuClose_activate, self)
		
		self.maximize()
		self.show_all()
	

File: Projects/Genetic Image Copy/mainwindow.cc
/*
This file is part of Fred Morcos' Genetic.

Fred Morcos' Genetic is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Fred Morcos' Genetic is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with Fred Morcos' Genetic.  If not, see <http://www.gnu.org/licenses/>.
*/

#include "mainwindow.h"

#include <QFileDialog>
#include <QDebug>

MainWindow::MainWindow(QWidget *parent) :
	QMainWindow(parent),
	m_simulation(new Simulation(this))
{
	setupUi(this);

	actionOpen->setIcon(QIcon::fromTheme("document-open"));
	actionQuit->setIcon(QIcon::fromTheme("application-exit"));
	actionStart->setIcon(QIcon::fromTheme("media-playback-start"));
	actionStop->setIcon(QIcon::fromTheme("media-playback-stop"));

	connect(m_simulation, SIGNAL(imageUpdated(QImage)),
			this, SLOT(onSimulationUpdate(QImage)));
	connect(qApp, SIGNAL(aboutToQuit()), m_simulation, SLOT(stop()));
}

void MainWindow::changeEvent(QEvent *e)
{
	QMainWindow::changeEvent(e);
	switch (e->type())
	{
	case QEvent::LanguageChange:
		retranslateUi(this);
		break;
	default:
		break;
	}
}

void MainWindow::on_actionOpen_triggered()
{
	QFileDialog *dialog = new QFileDialog(this);
	dialog->setFileMode(QFileDialog::ExistingFile);
	dialog->open(this, SLOT(onFileOpenDialogOpened(QString)));
}

void MainWindow::onFileOpenDialogOpened(const QString &file)
{
	labelOriginal->setPixmap(QPixmap(file));
}

void MainWindow::on_actionQuit_triggered()
{
	m_simulation->stop();
	qApp->quit();
}

void MainWindow::on_actionStop_triggered()
{
	m_simulation->stop();
}

void MainWindow::on_actionStart_triggered()
{
	if (!(labelOriginal->pixmap() == 0))
		m_simulation->start(labelOriginal->pixmap()->toImage());
}

void MainWindow::onSimulationUpdate(QImage img)
{
	labelCopy->setPixmap(QPixmap::fromImage(img));
	qApp->processEvents();
}
